spec/requests/posts_request_spec.rb

```diff
@@ -146,4 +146,59 @@
       end
     end
   end
+
+  describe '#destroy_post_request' do
+    let(:new_post) { create :post, user: }
+    image_path = File.join(File.dirname(__FILE__), '..', 'images', 'download.jpeg')
+    let(:user) { create :user }
+    let(:commentable) { create :comment, user:, commentable: new_post }
+    let(:comment) { create :comment, user:, commentable: }
+
+    it 'with no logged user - redirects to login' do
+      delete post_path(new_post)
+      expect(response).to redirect_to(new_user_session_path)
+    end
+
+    context 'with logged user' do
+      before do
+        user.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+        sign_in user
+        post post_likes_path(new_post),
+             params: { likeable_type: 'Post', likeable_id: new_post.id }
+        post post_likes_path(commentable),
+             params: { likeable_type: 'Comment', likeable_id: commentable.id }
+        post post_likes_path(comment),
+             params: { likeable_type: 'Comment', likeable_id: comment.id }
```

Comment:
We are not testing this, so let's just create them using active record helpers (for ex: `user.likes.create(likeable: new_post)`)